### ISSUE TITLE
bindToDomElement.text fails when the node has nested children 

### ISSUE THREAD

**Issue by afiore** [node-js-libs/node.io#2]
Hi there, and congratulations for the good work done so far with Node.io. 
I've been trying to re-use some of your clever [dom augmentation methods](https://github.com/chriso/node.io/blob/master/lib/node.io/dom.js) in my own screen scraping project. I believe these methods could come in handy even outside of node.io, as they at least provide a decent read only DOM API, which is currently lacking in both Tautologistics's html parser and Soupselect. However, I've noticed that the element's 'text' getter does not work when called with a node that has children elements. For instance, try selecting the header texts from [this DOM document](http://repubblica.it):

```
$('h3',document).each(function(item){   
    puts(item.text)
});

home/me/mycode/dom.js:173
    text = self.filter(text).entityDecode();
                ^
TypeError: Object #<an Object> has no method 'filter'
```

In your code, this is on line 225. My guess is that you have not fully implemented this bit yet. I would be happy to help you testing this. 

Best and thanks for sharing,
Andrea

**Comment by chriso** [comment]
Hi Andrea,
Thanks. Firstly, make sure you're using the latest version of node.io (0.2.0-1) - I fixed the "has no method 'filter'" error in an earlier release. 

As for the text method, it's probably a bit ambiguous at this stage, but the text method only grabs text from directly inside that tag, e.g. `<h3>Hello <b>World</b></h3>` will only return "Hello" if you select the H3 and use the `item.text` method. I have added another method to get around this of course.. use `item.fulltext` which also grabs the text from inside any children. I've tested this with repubblica.it ([code is here](https://gist.github.com/729582)).

I'll update the documentation soon so it's a bit more obvious, but for now the comments in `dom.js` have a bit more info:

```
 * Getters:
 *     - innerHTML
 *     - rawtext - returns text immediately inside the selected element
 *     - rawfulltext - same as rawtext, but also includes text inside children / nested elements
 *     - text - rawtext trimmed, BR's replaced with \n, and entities decoded
 *     - fulltext - rawfulltext trimmed, BR's replaced with \n, and entities decoded
```

Thanks again,
Chris
